dsaps/cli.py

```diff
@@ -69,15 +72,37 @@ def search(ctx, field, string, search_type):
               'collection.')
 @click.option('-n', '--coll_name', prompt='Enter the name of the collection',
               help='The name of the collection to be created.')
+@click.option('-m', '--metadata', prompt='Enter the path of the metadata file',
+              help='The path of the JSON file of metadata.')
+@click.option('-f', '--file_path', prompt='Enter the path',
+              help='The path of the content, a URL or local drive path.')
+@click.option('-t', '--file_type', prompt='Enter the file type',
+              help='The file type to be uploaded.')
+@click.option('-i', '--ingest_type', prompt='Enter the type of ingest',
+              help='The type of ingest to perform: local, remote.',
+              type=click.Choice(['local', 'remote']))
 @click.pass_context
-def newcoll(ctx, comm_handle, coll_name):
+def newcoll(ctx, comm_handle, coll_name, metadata, file_path, file_type,
+            ingest_type):
     client = ctx.obj['client']
-    coll_id = client.post_coll_to_comm(comm_handle, coll_name)
-    logger.info(coll_id)
-    # STEPS TO ADD
-    # post items to collections
-        # post bistreams to item_links
-        # post prov notes
+    start_time = ctx.obj['start_time']
+    with open(metadata, encoding='UTF-8') as fp:
+        coll_metadata = json.load(fp)
+        coll_id = client.post_coll_to_comm(comm_handle, coll_name)
+        file_dict = {}
+        if ingest_type == 'local':
+            files = glob.glob(f'{file_path}/**/*.{file_type}', recursive=True)
+            for file in files:
+                file_name = os.path.splitext(file)[0][file.rindex('/') + 1:]
```

Comment:
Are you just trying to get the name of the file, without the path, and without the extension? That's what it looks like. If so, use:
```python
os.path.splitext(os.path.basename(file))[0]
```

Reply:
Good catch, I had even used that function in models.py. Updated